### ISSUE TITLE
[Bug] Potential bug in VAE

### ISSUE THREAD

**Issue by zyksir** [Wan-Video/Wan2.2#230]
This is the code in `vae_2_1.py` and this code looks confusing. If `feat_cache[idx] != 'Rep'`, the code is actually doing repeative padding while if `feat_cache[idx] == 'Rep'`, it is doing zero padding. Is this a bug?

```python
if cache_x.shape[2] < 2 and feat_cache[
        idx] is not None and feat_cache[idx] != 'Rep':
    # cache last frame of last two chunk
    cache_x = torch.cat([
        feat_cache[idx][:, :, -1, :, :].unsqueeze(2).to(
            cache_x.device), cache_x
    ],
                        dim=2)
if cache_x.shape[2] < 2 and feat_cache[
        idx] is not None and feat_cache[idx] == 'Rep':
    cache_x = torch.cat([
        torch.zeros_like(cache_x).to(cache_x.device),
        cache_x
    ],
                        dim=2)
```

**Comment by GooThinker** [comment]
That's a fantastic question, and I appreciate you bringing it up as it gets to the core of the caching logic.
This entire `if` block implements the logic for updating the cache. It is designed to handle the `1+4*n` video frame structure used in the streaming process.

When `feat_cache[idx] == 'Rep'`, it signifies that we are processing the second chunk after having processed the initial single frame (the "image frame"). Because the first chunk had a length of only 1, the resulting feature map is too short to provide sufficient context. In this specific startup scenario, where `feat_cache[idx]` is still the placeholder `'Rep'`, we pad with zeros (`torch.zeros_like`). This approach is consistent with standard forward-padding practices for causal operations at the beginning of a sequence.

Subsequently, when `feat_cache[idx] != 'Rep'`, we are updating the cache for all subsequent chunks (the third, fourth, and so on). If the current output `cache_x` has fewer than 2 frames (i.e., `cache_x.shape[2] < 2`), it means the current chunk was too short to produce a complete context for the *next* chunk. To resolve this, we prepend the last frame from the *previous* chunk's cache, accessed via `feat_cache[idx][:, :, -1, :, :]`, to ensure the new cache entry is always of the required length.
